Microsoft.Research/Contracts/MsCorlib/System.Collections.Generic.IDictionary.cs

```diff
@@ -181,29 +183,31 @@ abstract class IDictionaryContract<TKey, TValue> : IDictionary<TKey,TValue>
 
     void IDictionary<TKey, TValue>.Add(TKey key, TValue value)
     {
+      Contract.Requires(!ReferenceEquals(key, null));
       // Contract.Requires(!ContainsKey(key));
       //modifies this.*;
       //Contract.Ensures(ContainsKey(key));
     }
 
-    bool IDictionary<TKey, TValue>.ContainsKey(TKey key)
+    public bool ContainsKey(TKey key)
```

Comment:
For a `ContractClass` it's a bit different, it's clear that all methods are interface implementations. Here I changed to implicit methods because it removed unnecessary complexity from some of the contracts.